plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/TokenFetchingMetastoreClientFactory.java

```diff
@@ -106,11 +122,22 @@ private DelegationToken getDelegationToken(String username)
 
     private DelegationToken loadDelegationToken(String username)
     {
-        try (ThriftMetastoreClient client = createMetastoreClient()) {
```

Comment:
got it.
may be I am missing something, with the above catch block, we are saying if the exception is `TException` than throw `TrinoException(HIVE_METASTORE_ERROR, e);` so it will not retry?

and based on [this](https://github.com/trinodb/trino/pull/21000#discussion_r1559890608), once we set 
```
.abortOn(MetaException.class)
```

than it will exit on the `MetaException`, so if `getDelegationToken()` throws `TException`, than it will be retried automatically?

Reply:
and one more thing, if we do not use the [stats.wrap](https://github.com/osscm/trino/blob/thrift-dt-retry/plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftMetastoreApiStats.java#L37-L67) than we won't be to publish the JMX metrics.

if you want we can just retry in this PR and JMX metrics in a separate PR. thanks

Reply:
updated.